pkg/proc/proc_linux.go

```diff
@@ -0,0 +1,85 @@
+package proc
+
+import (
+	"bufio"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"syscall"
+	"time"
+
+	"k8s.io/klog"
+)
+
+// parseProcForZombies parses the current procfs mounted at /proc
+// to find processes in the zombie state.
+func parseProcForZombies() ([]int, error) {
+	files, err := ioutil.ReadDir("/proc")
+	if err != nil {
+		return nil, err
+	}
+
+	var zombies []int
+	for _, file := range files {
+		processID, err := strconv.Atoi(file.Name())
+		if err != nil {
+			break
+		}
+		stateFilePath := filepath.Join("/proc", file.Name(), "status")
+		fd, err := os.Open(stateFilePath)
+		if err != nil {
+			klog.Errorf("Failed to open %q: %v", stateFilePath, err)
+			continue
+		}
+		defer fd.Close()
+		fs := bufio.NewScanner(fd)
+		for fs.Scan() {
+			line := fs.Text()
+			if strings.HasPrefix(line, "State:") {
+				if strings.Contains(line, "zombie") {
+					zombies = append(zombies, processID)
+				}
+				break
+			}
+		}
+	}
+
+	return zombies, nil
+}
+
+// StartReaper starts a goroutine to reap processes periodically if called
+// from a pid 1 process.
+// If period is less than 5 seconds, then it is set to 5 seconds.
```

Comment:
I would have it be only when 0, not when less than five.  If you have a very fast subprocess creating app, you might need to set it faster.